api/pkg/apis/v1alpha1/utils/utils_test.go

```diff
@@ -69,26 +69,53 @@ func TestFormatObjectEmptyDict(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Equal(t, "{}", string(val))
 }
+
 func TestFormatObjectDictJson(t *testing.T) {
 	obj := map[string]interface{}{
 		"foo": "bar",
+		"man": "what  ",
+		"can": "\"I\"",
+		"say": "?",
 	}
 	val, err := FormatObject(obj, false, "$.foo", "")
 	assert.Nil(t, err)
 	assert.Equal(t, "\"bar\"", string(val))
+	val, err = FormatObject(obj, false, "$.man", "json")
```

Comment:
If I remember correctly, we have some issues in current object to YAML function. It cannot deal with YAML conversion in all cases. We need to check whether there's YAML go package and replace the logic in FormatObject.

Reply:
I replaced it with `gopkg.in/yaml.v2` in the new version

Reply:
I refactored the `FormatObject` function. The logic about "marshal obj into a json, and then unmashal back" is only for the input `obj` is an array. I moved it to the `isArray` condition branch.